Client/game_sa/CPoolsSA.cpp

```diff
@@ -377,82 +335,43 @@ CObject* CPoolsSA::AddObject(DWORD dwModelID, bool bLowLod, bool bBreakingDisabl
     return pObject;
 }
 
-void CPoolsSA::RemoveObject(unsigned long ulID, bool)
-{
-    DEBUG_TRACE("void CPoolsSA::RemoveObject ( unsigned long ulID, bool )");
-
-    static bool bIsDeletingObjectAlready = false;            // to prevent delete being called twice
-    if (!bIsDeletingObjectAlready && ulID != INVALID_POOL_ARRAY_ID)
-    {
-        bIsDeletingObjectAlready = true;
-
-        CObjectSA* pObjectSA = m_objectPool.array[ulID];
-        assert(NULL != pObjectSA);
-
-        // Pop the element to remove from the pool array
-        if (ulID != m_objectPool.ulCount - 1)
-        {
-            // We are removing an intermediate position of
-            // the array. Move the last element to the just
-            // deleted element position to not allow empty
-            // spaces on it.
-            m_objectPool.array[ulID] = m_objectPool.array[m_objectPool.ulCount - 1];
-            m_objectPool.array[ulID]->SetArrayID(ulID);
-        }
-        m_objectPool.array[m_objectPool.ulCount - 1] = NULL;
-
-        // Unlink the element to remove from the pool map
-        objectPool_t::mapType::iterator iter = m_objectPool.map.find(pObjectSA->GetObjectInterface());
-        if (iter != m_objectPool.map.end())
-        {
-            m_objectPool.map.erase(iter);
-        }
-
-        // Delete it from memory
-        delete pObjectSA;
-
-        // Decrease the count of elements in the pool
-        --m_objectPool.ulCount;
-
-        bIsDeletingObjectAlready = false;
-    }
-}
-
 void CPoolsSA::RemoveObject(CObject* pObject, bool bDelete)
 {
     DEBUG_TRACE("void CPoolsSA::RemoveObject ( CObject* pObject, bool bDelete )");
 
     assert(NULL != pObject);

```

Comment:
Add this check here:
```
static bool bIsDeletingObjectAlready = false;            // to prevent delete being called twice
if (!bIsDeletingObjectAlready)
{
    bIsDeletingObjectAlready = true;
```